ErrorMessageBestPractices.md

```diff
@@ -0,0 +1,72 @@
+# Guidelines for Writing Effective Error Messages ✍️
+It's important to remember the impact that clear and informative error messages can have on debugging and maintenance. A well-crafted error message can save hours of troubleshooting and make our codebase more user-friendly, especially for those who may be less familiar with the inner workings of the system.
+
+## Common Mistakes
+1. **Being Too Vague:** Avoid generic messages like "An error occurred" or "Invalid input." These do not help in identifying the root cause.
+2. **Not Including Variable Values:** Forgetting to include the actual values that caused the error can make debugging significantly harder.
+3. **Overloading Messages:** While it’s important to be informative, avoid making the message too long or cluttered. Keep it concise and to the point.
+4. **Assuming Prior Knowledge:** Don’t assume that the person encountering the error knows as much about the code as you do. Provide enough context to make the error understandable on its own.
+5. **Not Making the Message Actionable:** Ensure that the error message provides clear guidance on what needs to be done to resolve the issue. Simply stating what went wrong without suggesting how to fix it can leave the user frustrated and confused.
+
```

Comment:
What do you think about including a line-item guidance for `TT_FATAL` vs. `TT_ASSERT`. I've hit into issues where it should've been `TT_FATAL` but it was written with `TT_ASSERT` and it didn't error out because it was compiled in release-mode.

Reply:
👍